### DISCUSSION TITLE
Theme for a custom vue component

### DISCUSSION THREAD

**Discussion by DrakoPOD** [vuejs/vitepress#1371] [Q&A]
I'm trying to make a component tab, for code code block:
![image](https://user-images.githubusercontent.com/12172698/191081930-d826a4e1-73a0-40f7-b358-9a42ee74b766.png)
How I can use the default theme of the page? For example, when I use `button`, is appear as simple text, only the mouse point have hovere effect.

Does vitepress has a list of css tags? like default colors, also, access to theme variable for light and dark mode.

**Comment by brc-dd** [comment]
https://github.com/vuejs/vitepress/blob/main/src/client/theme-default/styles/vars.css

**Reply by erondpowell** [reply]
@brc-dd I feel like there might be a "style bug" here 🤔 

I'm trying to build the exact same type of component right now, buttons to toggle display between code/API snippets. 
On a custom component, the full Vitepress theme css (as well as our project's overrides.css) are all inherited....

*Except the button*. The button was sanitized of even default HTML styling. (I added that green background myself)

---
<img width="756" alt="Screen Shot 2022-09-22 at 3 28 28 PM" src="https://user-images.githubusercontent.com/62306630/191844775-1d048c52-de00-4cd6-897e-0cb20e420e66.png">

---

![Screen Shot 2022-09-22 at 3 39 07 PM](https://user-images.githubusercontent.com/62306630/191846588-5b96d05e-42e3-4f64-be6c-48c23b942573.png)

----

Is this the original intention with the css? or am I missing something? 

Thanks! 🙏

**Reply by brc-dd** [reply]
> Except the button. The button was sanitized of even default HTML styling. (I added that green background myself)

There is nothing like "default HTML styling". You probably are referring to useragent stylesheets which make button look like a weird box. Those won't be applied as we are [resetting the css](https://en.wikipedia.org/wiki/Reset_style_sheet).

Also, we haven't done any styling to buttons. I mean all those button styles are scoped and won't be applied to the buttons you add using md/vue. You need to style them manually. So what you're seeing is expected behavior.

Here is our button component BTW: https://github.com/vuejs/vitepress/blob/main/src/client/theme-default/components/VPButton.vue, you can copy, modify and use it for your buttons.

**Reply by erondpowell** [reply]
@brc-dd  Thanks for the reply!

Yes, I was talking about useragent style sheets.

By the way, my little "toggle API Style" component is almost done. 
If there is an appropriate place to share it, I'm happy to do that.
I would guess toggling ~~API styles~~ content is a fairly common use case.

**Reply by brc-dd** [reply]
> If there is an appropriate place to share it, I'm happy to do that.

We have a Discord channel at https://chat.vuejs.org 🙌

**Reply by erondpowell** [reply]
Sweet, I'll post a link whenever its ready.
